src/main/java/org/broadinstitute/hellbender/utils/read/markduplicates/sparkrecords/Pair.java

```diff
@@ -55,21 +55,36 @@ public Pair(final GATKRead read1, final GATKRead read2, final SAMFileHeader head
         final int read1UnclippedStart = ReadUtils.getStrandedUnclippedStart(read1);
         final int read2UnclippedStart = ReadUtils.getStrandedUnclippedStart(read2);
 
-        if( read1UnclippedStart < read2UnclippedStart ){
+        int read1ReferenceIndex = ReadUtils.getReferenceIndex(read1,header);
+        int read2ReferenceIndex = ReadUtils.getReferenceIndex(read2,header);
+
+        if( read1ReferenceIndex != read2ReferenceIndex ? read1ReferenceIndex < read2ReferenceIndex : read1UnclippedStart <= read2UnclippedStart ){
             first = read1;
             second = read2;
         } else {
             first = read2;
             second = read1;
         }
-        // Keep track of the orientation of read1 and read2 as it is important for optical duplicate marking
-        wasFlipped = second.isFirstOfPair();
-
         firstStartPosition = first.getAssignedStart();
 
+        // Ensuring that the orientation is consistent for reads starting at the same place there is no propper order
```

Comment:
I think the comment in picard is clearer

Reply:
// if the two read ends are in the same position, pointing in opposite directions,
                            // the orientation is undefined and the procedure above
                            // will depend on the order of the reads in the file.
                            // To avoid this, we set it explicitly (to FR):

Reply:
also, propper -> proper